main/lsp/requests/rename.cc

```diff
@@ -114,15 +241,24 @@ unique_ptr<ResponseMessage> RenameTask::runRequest(LSPTypecheckerDelegate &typec
             auto resp = move(queryResponses[0]);
             // Only supports rename requests from constants and class definitions.
             if (auto constResp = resp->isConstant()) {
+                // Sanity check the text.
+                if (islower(params->newName[0])) {
```

Comment:
This check is also going to be needed if `resp->isDefinition()` and the symbol `isClassOrModule()`.

This reminds me that _method_ renames have to be lower case. You can choose if you want to add that check now or later. :)